app/controllers/validation/package_query_parameters.rb

```diff
@@ -0,0 +1,41 @@
+# frozen_string_literal: true
+
+# rubocop:disable Naming/VariableName
+
+module Validation
+  class PackageQueryParameters
+    include ActiveModel::Validations
+
+    attr_accessor :sortFilter, :selectedFilter,
+                  :contentTypeFilter
+
+    validates :sortFilter, inclusion: { in: %w[name relevance],
+                                        message: 'Invalid Query Parameter for sort' }, allow_nil: true
+    validates :selectedFilter, inclusion: { in: %w[true false ebsco],
+                                            message: 'Invalid Query Parameter for filter[:selected]' }, allow_nil: true
+    validates :contentTypeFilter, inclusion: { in: %w[aggregatedfulltext abstractandindex ebook ejournal print onlinereference unknown],
+                                               message: 'Invalid Query Parameter for filter[:type]' }, allow_nil: true
+
+    def initialize(params = {})
+      filters = params.fetch(:filter, nil)
+      @sortFilter = params[:sort]
+      @selectedFilter = hash?(filters) ? get_filter(filters, :selected) : filters
+      @contentTypeFilter = hash?(filters) ? get_filter(filters, :type) : filters
+    end
+
+    private
+
+    def get_filter(hash, key)
```

Comment:
thiiis might be overkill. i'm all for single responsibility principle but hash already has the responsibility of doing hash lookups so why not just hit it head-on whence this method is invoked.